### PR TITLE
fix regex for end of line

### PR DESCRIPTION
**Pull request by ksami**

### CODE REVIEW

**Comment by PRMerger9** [comment]
@ksami : Thanks for your contribution! The author, @gewarren, has been notified to review your proposed change.

File: docs/ide/using-regular-expressions-in-visual-studio.md [hunk 1]
```diff
@@ -42,8 +42,8 @@ Here are some examples:
 |Match any character one or more times (Wildcard ?)|.+|`e.+e` matches "eede" in "feeder" but not "ee".|
 |Match zero or more occurrences of the preceding expression (match as few characters as possible)|*?|`e.*?e` matches "ee" in "feeder" but not "eede".|
 |Match one or more occurrences of the preceding expression (match as few characters as possible)|+?|`e.+?e` matches "ente" and "erprise" in "enterprise", but not the whole word "enterprise".|
-|Anchor the match string to the beginning of a line or string|^|`^car` matches the word "car" only when it appears at the beginning of a line.|
-|Anchor the match string to the end of a line|\r?$|`End\r?$` matches "end" only when it appears at the end of a line.|
+|Anchor the match string to the beginning of a line|^|`^car` matches the word "car" only when it appears at the beginning of a line.|
+|Anchor the match string to the end of a line|$|`end$` matches the word "end" only when it appears at the end of a line.|
```
Comment (gewarren) [inline docs/ide/using-regular-expressions-in-visual-studio.md]: @ksami This is not the behavior I'm seeing in Visual Studio. Is this perhaps for a different flavor of regular expressions? The \r?$ is what matches the end of a line in Visual Studio.

Comment (ksami) [reply docs/ide/using-regular-expressions-in-visual-studio.md]: Nope, I'm going by this https://docs.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-language-quick-reference#anchors. `\r?$` does work, the `\r?` seemed unnecessary for me but I now realise I'm using Visual Studio for Mac.

I have to note though that anchoring to the end of the line actually anchors to the end of the file. Could the description or regex be changed to either make this clearer or anchor to each end of line instead?

Comment (gewarren) [reply docs/ide/using-regular-expressions-in-visual-studio.md]: @ksami I tried it out and you're right, $ matches the end of the file. I can add a row to the table for that.

To match the end of a line in Visual Studio (on the Windows version at least), one does need the full \r?$ syntax:

image

**Pull request closed without merging**